src/jit/codegenxarch.cpp

```diff
@@ -5431,7 +5420,7 @@ void CodeGen::genCallInstruction(GenTreeCall* call)
             assert(curArgTabEntry != nullptr);
             assert(size == (curArgTabEntry->numSlots * TARGET_POINTER_SIZE));
 #ifdef FEATURE_PUT_STRUCT_ARG_STK
-            if (source->TypeGet() == TYP_STRUCT)
+            if (!source->OperIs(GT_FIELD_LIST) && (source->TypeGet() == TYP_STRUCT))
```

Comment:
This is the scary part of the change - the second commit. `GT_FIELD_LIST` (the head) had the same type as the first field in the list - a primitive, non-struct type, of course. I've made it `TYP_STRUCT` because it was just misleading before. 

Unfortunately there were a couple of places that checked for struct typed nodes in such a way that now need to be updated - basically they assumed that if it's `TYP_STRUCT` it's not `FIELD_LIST`.

I hope I caught them all, both by inspecting a bunch of code that checked for `GT_FIELD_LIST` and by relying on asserts.